javascript/javascript1/week2/jsweek2.js

```diff
@@ -0,0 +1,103 @@
+// Flight booking fullname function
+function getFullname(firstname, surname, useFormalName) {
+    if (useFormalName) {
+        return "Lord " + firstname + " " + surname;
+    } else {
+        return firstname + " " + surname;
+    }
+}
+const fullname1 = getFullname("Benjamin", "Hughes", true);
+const fullname2 = getFullname("Benjamin", "Hughes", false);
+
+console.log(fullname1);
+console.log(fullname2);
+
+
+
+// Event application
+const wDays7 = [
+    "Thursday",
+    "Friday",
+    "Saturday",
+    "Sunday",
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+];
+function getEventWeekday(daysOfEvent) {
+    const fullD = new Date();
+    const manyDaysFromToday = new Date(
+        fullD.getDate(),
+        fullD.getMonth() + daysOfEvent
+    );
+    return wDays7[manyDaysFromToday.getDay()];
+}
+// With todays weekday a tuesday
+console.log(getEventWeekday(9)); // Logs out "Thursday"
+
+// With todays weekday a Friday
+console.log(getEventWeekday(5)); // Logs out "Sunday"
+
+
+
+// Weather wear
+function weatherWear(temperature) {
+    if (temperature < 30) {
+        temperature = "Jaket in cold weather";
+    } else {
+        temperature = "T-shirt in hot weather";
+    }
+    return `Hi you should wearing a ${temperature}`;
+}
+console.log(weatherWear(10));
+
+
+
+// Student manager
+const class07Students = [];
+function addStudentToClass(studentName) {
+    if (class07Students.length > 6) {
+        console.log("Cannont add more students to class 07");
+    } else if (class07Students.includes(studentName)) {
+        console.log("studen" + studentName + "is already in the class 07.");
+    } else if (studentName === "Queen") {
+        class07Students.push(studentName);
+    } else if (studentName === "") {
+        console.log("You cannot add an studen without name,please add name.");
+    } else {
+        class07Students.push(studentName);
+    }
+}
+function getNumberOfStudents() {
+    return class07Students.length;
+}
+addStudentToClass("");
+addStudentToClass("ali");
+addStudentToClass("ali");
+addStudentToClass("Alex");
+addStudentToClass("Nine");
+addStudentToClass("Susan");
+addStudentToClass("Marco");
+addStudentToClass("Vital");
+addStudentToClass("Queen");
+console.log(getNumberOfStudents());
+
+
+
+// addCandy function
+
+function addCandy(candyt, pricepergm) { }
+addCandy("Chocolate", 100);
+addCandy("Gummies", 50);
+addCandy("Lollipops", 30);
+
+const amountToSpend = Math.random(1) * 100;
+
+function canBuyMoreCandy() {
+    if (amountToSpend > 9) {
+        console.log("You can buy more, so please do!");
+    }
+    console.log("Enough candy for you!");
+}
+canBuyMoreCandy();
+
```

Comment:
Super cool, that you worked with the optional assignment as well. 💪🏼
This is a great starting point for the solution. 👌
There are some missing elements before you reach the finish line 🏁

In the current implementation, the addCandy function is empty. 
- According to the assignment description, the function should add the price of the candy to an array called boughtCandyPrices using push
- Where could you declare the array boughtCandyPrices?
- How you could push the price of the candy to this array? 🧐

Looking at the assignment, you could consider storing the candy and the prices in a nested array. Something like 
const candyPrices = [[Sweet, 0.5], [Chocolate, 0.7]] etc.
- There are many other ways to do this, so consider how/or if you would like to store the candy values.

In the function canBuyMoreCandy, you are on the right track in terms of using the if else structure
- Think about how you can sum the price of the candy that has been pushed to the array boughtCandyPrices. Maybe that takes your implementation in a different direction. 💡

Let me know, if you would like to talk it through on Slack. 😊
You have done really well in this homework overall, great work 🔥